### DISCUSSION TITLE
Improve handling of session timeout and logged out when still in the interface

### DISCUSSION THREAD

**Discussion by adam-jones-net** [area17/twill#1009] [Ideas]
If you are in the admin, especially on the add/edit record page, its very easy for ajax/vue content to still function and suggest that all is well when in fact it isn't.

For example, if you are in an edit page and get distracted and later return to your edit, you have no idea the session has timed out.   The user carries on completing the fields and when they save/update they just get a generic msg in a red error at the bottom of the page stating that there was a problem saving the record.    It's not clear it's because of the session and of course they have likely lost their input now as they will have to redirect manually to the login again.

Also when editing and you open a browser field, the linked records that would normally show won't; the list will just be empty.     It appears that there is no check of the session so therefore no msg appears, which I would suggest it should produce. 

A more advanced solution could be to check if the session can be restored.     You see that kind of functionality in the Spectrum chat system when a little msg pops up reporting 'restoring session' or something to that effect.

**Comment by ifox** [comment]
If your session actually reach the timeout, it is supposed to show you a different message that invite you to login again in a different tab, so that you can continue working. I will try to reproduce, it might be a regression, see:
https://github.com/area17/twill/blob/09e22483554b574c0137c824829ac5fedc2cd33e/frontend/js/utils/errors.js#L14
